atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTableTest.java

```diff
@@ -0,0 +1,172 @@
+import com.palantir.nexus.db.DBType;
+import com.palantir.nexus.db.sql.AgnosticResultSetImpl;
+import com.palantir.nexus.db.sql.SqlConnection;
+import com.palantir.remoting.api.config.service.HumanReadableDuration;
+
+public class PostgresDdlTableTest {
+    private PostgresDdlTable postgresDdlTable;
+    private static final ConnectionSupplier connectionSupplier = mock(ConnectionSupplier.class);
+    private static final TableReference TEST_TABLE = TableReference.createFromFullyQualifiedName("ns.test");
+
+    private static final long NOW_MILLIS = 1000;
+    private static final long COMPACT_INTERVAL_MILLIS = 100;
+    private static final long SMALL_POSITIVE_FACTOR = 10;
+
+    @Before
+    public void setUp() {
+        postgresDdlTable = new PostgresDdlTable(TEST_TABLE,
+                connectionSupplier,
+                ImmutablePostgresDdlConfig.builder()
+                        .compactInterval(HumanReadableDuration.milliseconds(COMPACT_INTERVAL_MILLIS))
+                        .build());
+    }
+
+    @Test
+    public void shouldCompactIfVacuumWasNeverPerformedAndTheDbTimeIsLessThanCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(null, COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR);
+
+        assertThatVacuumWasPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldCompactIfVacuumWasNeverPerformedAndTheDbTimeIsMoreThanCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(null, COMPACT_INTERVAL_MILLIS * SMALL_POSITIVE_FACTOR);
+
+        assertThatVacuumWasPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldCompactIfVacuumWasPerformedExactlyBeforeCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS, NOW_MILLIS);
+
+        assertThatVacuumWasPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldCompactIfVacuumWasPerformedBeforeCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS * SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
+
+        assertThatVacuumWasPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldNotCompactIfVacuumWasPerformedWithinCompactInterval() {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS - COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
+        assertThatVacuumWasNotPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldNotCompactIfVacuumTimestampExceedsNowTimestampByLessThanCompactInterval() {
+        SqlConnection sqlConnection = setUpSqlConnection(NOW_MILLIS + COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR,
+                NOW_MILLIS);
+        assertThatVacuumWasNotPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldCompactIfVacuumTimestampExceedsNowTimestampByMoreThanCompactInterval() {
```

Comment:
should**Not**CompactIfVacuumTimestampExceedsNowTimestampByMoreThanCompactInterval

I spent 5 minutes trying to figure out how this is passing before seeing that we assert it was not performed :P

Reply:
Oops :( I just copied this from the original so this should also be fixed on develop.